build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/internal/core/jsoncdb/generator/CompilationDatabaseGenerator.java

```diff
@@ -229,10 +229,10 @@ private List<CompilationDatabaseInformation> populateObjList(IProject project, I
 						outputLocation + "", inputStrings, sourceLocation, outputLocation); //$NON-NLS-1$
 
 				IBuildMacroProvider provider = ManagedBuildManager.getBuildMacroProvider();
-				String compilerName = CompilationDatabaseGenerator.getCompilerName(tool);
 				String commandLine = cmdLInfo.getCommandLine();
+				String compilerName = CompilationDatabaseGenerator.getCompilerName(commandLine);
```

Comment:
The replace on the next line does not work reliabliy. Please revisit this bit of code. For example, if commandLine has this contents:

```
"name of compiler" -O0 input.c -i output.o
```

Then `compilerName` will be `name of compiler` and after the replace the command line will be:

```
"" -O0 input.c -i output.o
```

which leads to, `resolvedOptionFileContents` being:

```
"/full/path/to/name of compiler" "" -O0 input.c -i output.o
```

Instead of doing `replace` use the parsed information of `CommandLineUtil.argumentsToArray` to convert the `commandLine` into two parts, the compilerName (i.e. argv[0]) and the rest of the command line as a string.